virttest/libvirtd_decorator.py

```diff
@@ -0,0 +1,129 @@
+"""
+A decorator utility functions to apply libvirtd functions.
+
+Copyright: Red Hat Inc. 2020
+"""
+import logging
+import os
+import re
+from virttest import data_dir
+
+from avocado.utils import process
+from avocado.utils import path
+from avocado.utils import astring
+
+
+try:
+    path.find_command("libvirtd")
+    LIBVIRTD = "libvirtd"
+except path.CmdNotFoundError:
+    LIBVIRTD = "virtqemud"
+
+
+def get_libvirtd_split_enable_bit():
+    base_cfg_path = os.path.join(data_dir.get_shared_dir(), 'cfg', 'base.cfg')
+    if os.path.isfile(base_cfg_path):
+        with open(base_cfg_path, 'r') as base_file:
+            for line in base_file:
+                if 'enable_split_libvirtd_feature' in line and 'yes' in line:
+                    return True
+    else:
+        logging.info("CAN NOT find base.cfg file")
+    return False
+
+
+def get_libvirt_version_compare(major, minor, update, session=None):
+    """
+    Determine/use the current libvirt library version on the system
+    and compare input major, minor, and update values against it.
+    If the running version is greater than or equal to the input
+    params version, then return True; otherwise, return False.
+
+    This is designed to handle upstream version comparisons for
+    test adjustments and/or comparisons as a result of upstream
+    fixes or changes that could impact test results.
+
+    :param major: Major version to compare against
+    :param minor: Minor version to compare against
+    :param update: Update value to compare against
+    :param session: Shell session on remote host
+    :return: True if running version is greater than or
+                  equal to the input libvirt version
+    """
+    LIBVIRT_LIB_VERSION = 0
+
+    func = process.system_output
+    if session:
+        func = session.cmd_output
+
+    libvirt_ver_cmd = "%s -V" % LIBVIRTD
+    logging.warning("livirt_ver_cmd")
+    logging.warn(libvirt_ver_cmd)
+    try:
+        regex = r'%s\s*.*[Ll]ibvirt.*\s*' % LIBVIRTD
+        regex += r'(\d+)\.(\d+)\.(\d+)'
+        lines = astring.to_text(func(libvirt_ver_cmd)).splitlines()
+        logging.warning("libvirt version value by libvirtd or virtqemud command: %s" % lines)
+        for line in lines:
+            mobj = re.search(regex, line.strip())
+            if bool(mobj):
+                LIBVIRT_LIB_VERSION = int(mobj.group(1)) * 1000000 + \
+                                      int(mobj.group(2)) * 1000 + \
+                                      int(mobj.group(3))
+                break
+    except (ValueError, TypeError, AttributeError):
+        logging.warning("Error determined libvirt version")
+        return False
+
+    compare_version = major * 1000000 + minor * 1000 + update
+    if LIBVIRT_LIB_VERSION >= compare_version:
+        return True
+    return False
+
+
+LIBVIRTD_SPLIT_ENABLE_BIT = get_libvirtd_split_enable_bit()
+IS_LIBVIRTD_SPLIT_VERSION = get_libvirt_version_compare(5, 6, 0)
```

Comment:
I believe this version now will fallback and not crash if libvirt is not installed, but I'm not 100% sure because the Cirrus CI is failing, and it shouldn't if libvirtd is not *required* to simply load Avocado-VT.  I need to test it on a live system though.

Reply:
Issue was resolved to avoid use libvirtd command if not exist